### DISCUSSION TITLE
How do I change the public key of a RustDesk pro server?

### DISCUSSION THREAD

**Discussion by mitchellarganbright** [rustdesk/rustdesk#5790] [Q&A] [answered]
I recently upgraded to the rustdesk pro server, but when i installed it the public key changed from what it was before. I can't connect to any of my client computers now. I've tried to edit id_ed25519.pub but upon restart it always resets to what it came with. 

The web console also doesn't give any option to edit the public key, only the private key. 

Any help would be appreciated, thanks.

**Comment by dinger1986** [comment]
Stop the service, delete the files and start again and it will create a new key etc

**Reply by mitchellarganbright** [reply]
Yes but I need to change the key to the old one that all my client's are already configured to use.

**Reply by dinger1986** [reply]
Do you still have the old one? If so copy it in

**Reply by mitchellarganbright** [reply]
It's not taking the public key I copy into id_ed25519.pub even after i restart the service
![image](https://github.com/rustdesk/rustdesk/assets/13736705/9c88afe9-81ad-44a8-ba84-a2d0c4d408b6)

**Reply by dinger1986** [reply]
Need the private key as well

**Comment by rustdesk** [comment]
Or copy your old private key here.

![image](https://github.com/rustdesk/rustdesk/assets/71636191/572ec13f-9551-4406-b803-504b2af26aa0)

**Reply by mitchellarganbright** [reply]
I don't have the old private key from before I updated the server. Where would I find that?

**Reply by dinger1986** [reply, marked as answer]
It depends how you installed before, but if you don't have the private key it'll not work

**Reply by mitchellarganbright** [reply]
Thanks for your help!
